frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/SlideOutCurrentBranchAction.java

```diff
@@ -2,12 +2,30 @@
 
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
 import com.virtuslab.gitmachete.frontend.actions.base.BaseSlideOutBranchAction;
 
 public class SlideOutCurrentBranchAction extends BaseSlideOutBranchAction {
   @Override
   public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
     return getCurrentBranchNameIfManaged(anActionEvent);
   }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    var isEnabledAndVisible = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn))
+        .flatMap(b -> Option.of(b.isNonRootBranch() ? b.asNonRootBranch() : null))
+        .map(nrb -> nrb.getSyncToParentStatus() == SyncToParentStatus.MergedToParent)
+        .getOrElse(false);
+
+    anActionEvent.getPresentation().setEnabledAndVisible(isEnabledAndVisible);
+
+    if (isEnabledAndVisible) {
+      super.onUpdate(anActionEvent);
```

Comment:
Hmm is it okay to NOT call `super.onUpdate` in certain cases (CC @micpiotrowski)?
* If never okay, pls rework + possibly update scripts/enforce-call-super-for-overridden-onupdate.
* Either way, pls document the requirements in `com.virtuslab.gitmachete.frontend.actions.base.BaseProjectKeyAvailabilityAssuranceAction#onUpdate` javadoc

Reply:
Whoops it was not okay since. We should call it always and b4 any other things since it is implemented by `BaseGitMacheteRepositoryReadyAction` (among others). Refactored.